### DISCUSSION TITLE
DONGLE_SERIAL_NUMBER

### DISCUSSION THREAD

**Discussion by Pablo-or-Pablo** [kellerza/sunsynk#174] [General]
Hi,
I cannot find DONGLE_SERIAL_NUMBER for my /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_20afef646093eb11b8a11b4f3d98b6d1-if00-port0
Still "invalid literal for int() with base 10".

What I should insert in this var?
![image](https://github.com/kellerza/sunsynk/assets/74837976/e7add56c-d912-4a86-952d-0d24601b766f)

**Comment by kellerza** [comment]
You only need dongle_serial_number for the solarman dongle/driver - keep it an empty string “”

for usb, use the pymodbus driver.

**Reply by Pablo-or-Pablo** [reply]
Yes, I need it cause I want to use solarman.
I don't have any adaptor for this inwerter.

**Reply by kellerza** [reply]
The serial number is printed on a sticker on the side of the dongle

**Reply by Pablo-or-Pablo** [reply]
Thank you. It works, but something is still wrong, but I don't know what:

ConnectionRefusedError: [Errno 111] Connect call failed ('192.168.1.182', 502)

This IP is from my logger in Deye Inverter.

**Reply by kellerza** [reply]
Have you tried port 8899?

@Xhanti , any suggestions?

**Reply by Xhanti** [reply]
Hi,
Try:

1. Visit http://**IP**/config_hide.html (where **IP** is your solarman dongle's ip, you should be able to get that from your router) 
2. Login in with username: admin,  password: admin (the defauls)
3. Scroll down until you see **Internal Server Parameters** setting
4. Check the port, it should be 8899 as @kellerza said. If its not, use what's there.
<img width="420" alt="image" src="https://github.com/kellerza/sunsynk/assets/2096304/604f538f-b96e-4c1d-b643-221513e3689c">


I would not change any of the settings there, I have no idea what that would do but I don't think it would be good. 
If you still don't come right, can you turn on debugging in the addon and show us the output so we can see where it failed?

The connection bits inside the home assistant addon should look something like this:

DONGLE_SERIAL_NUMBER: "DONGLESERIALNUMBER" (Not to be confused with the inverter serial number)
PORT: tcp://IP:8899 (Your dongle's IP address)

**Reply by Pablo-or-Pablo** [reply]
Write. I had a bad port.
There should by stable bonnection, but I have now another problem:
"Frame error retry attempt(4): V5 frame contains incorrect data logger serial number"

**Reply by Pablo-or-Pablo** [reply]
After 20 iterations there is:

OSError: OSError reading 5 registers from 3: Failed to read register 3; OSError reading 2 registers from 20: Failed to read register 20; OSError reading 1 registers from 117: Failed to read register 117; OSError reading 6 registers from 148: Failed to read register 148
17:39:02 ERROR   Task was destroyed but it is pending!
task: <Task pending name='ConnKeeper' coro=<PySolarmanV5Async._conn_keeper() running at /usr/local/lib/python3.11/site-packages/pysolarmanv5/pysolarmanv5_async.py:139> wait_for=<Future pending cb=[Task.task_wakeup()]>>
[17:39:02] WARNING: Halt add-on with exit code 1

**Reply by Xhanti** [reply]
@Pablo-or-Pablo I've seen that when the serial number of the inverter, not the dongle is put in the  DONGLE_SERIAL_NUMBER field. The 2 numbers look very similar but are distinct.  If you navigate to: http://IP/index_cn.html you will see the dongle serial number on your left.

<img width="794" alt="image" src="https://github.com/kellerza/sunsynk/assets/2096304/a1622876-1d5d-41c9-9ebd-57c9b20c036d">

**Reply by Xhanti** [reply]
@Pablo-or-Pablo it looks like its connected, its the sensors it doesn't like. Can we see your yaml file without your serial number or mqtt password (if specified)?

**Reply by Pablo-or-Pablo** [reply]
I was trying with the serial of inverter and logger (Solarman add-on is using number of logger), but the result is the same.
Full logs:

```log
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun sunsynk (no readiness notification)
s6-rc: info: service legacy-services successfully started
17:52:37 INFO    Loading HASS OS configuration
17:52:37 INFO    Schedules:
+-----------+-----+------+--------+-----------+----------+------------+
|    Key    | src | Read | Report | Change by | Change % | Change any |
+-----------+-----+------+--------+-----------+----------+------------+
| date_time |     |  60  |   60   |           |          |    True    |
|     rw    |     |  5   |  300   |           |          |    True    |
|     w     |  *  |  10  |   60   |     80    |          |            |
|    kwh    |     | 300  |  300   |           |          |            |
|  any_unit |     |  15  |  300   |           |          |            |
|  no_unit  |     |  15  |  300   |           |          |    True    |
+-----------+-----+------+--------+-----------+----------+------------+
17:52:37 INFO    Using three phase sensor definitions.
17:52:37 ERROR   Unknown sensor specified: aux_power
17:52:37 ERROR   Unknown sensor specified: essential_power
17:52:37 ERROR   Unknown sensor specified: grid_connected
17:52:37 ERROR   Unknown sensor specified: inverter_current
17:52:37 ERROR   Unknown sensor specified: load_frequency
17:52:37 ERROR   Unknown sensor specified: non_essential_power
17:52:37 INFO    Added hidden sensors as other sensors depend on it: Rated power, Serial
17:52:37 INFO    Connecting to tcp://192.168.1.182:8899
17:52:37 INFO    Reading startup sensors Prog4 Time, Prog1 Time, Battery Low Capacity, Prog6 Time, Rated power, Serial, Prog5 Time, Prog3 Time, Prog2 Time
17:52:37 INFO    Frame error retry attempt(0): V5 frame contains incorrect data logger serial number
17:52:39 INFO    Frame error retry attempt(1): V5 frame contains incorrect data logger serial number
17:52:41 INFO    Frame error retry attempt(2): V5 frame contains incorrect data logger serial number
17:52:44 INFO    Frame error retry attempt(3): V5 frame contains incorrect data logger serial number
17:52:46 INFO    Frame error retry attempt(4): V5 frame contains incorrect data logger serial number
17:52:48 INFO    Frame error retry attempt(0): V5 frame contains invalid sequence number
17:52:50 INFO    Frame error retry attempt(1): V5 frame contains incorrect data logger serial number
17:52:52 INFO    Frame error retry attempt(2): V5 frame contains incorrect data logger serial number
17:52:54 INFO    Frame error retry attempt(3): V5 frame contains invalid sequence number
17:52:56 INFO    Frame error retry attempt(4): V5 frame contains incorrect data logger serial number
17:52:58 INFO    Frame error retry attempt(0): V5 frame contains incorrect data logger serial number
17:53:00 INFO    Frame error retry attempt(1): V5 frame contains incorrect data logger serial number
17:53:03 INFO    Frame error retry attempt(2): V5 frame contains incorrect data logger serial number
17:53:05 INFO    Frame error retry attempt(3): V5 frame contains incorrect data logger serial number
17:53:08 INFO    Frame error retry attempt(4): V5 frame contains incorrect data logger serial number
17:53:09 INFO    Frame error retry attempt(0): V5 frame contains incorrect data logger serial number
17:53:11 INFO    Frame error retry attempt(1): V5 frame contains incorrect data logger serial number
17:53:13 INFO    Frame error retry attempt(2): V5 frame contains incorrect data logger serial number
17:53:16 INFO    Frame error retry attempt(3): V5 frame contains incorrect data logger serial number
17:53:18 INFO    Frame error retry attempt(4): V5 frame contains incorrect data logger serial number
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/__main__.py", line 72, in <module>
    main()
  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/__main__.py", line 67, in main
    loop.run_until_complete(main_loop(loop))
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/__main__.py", line 37, in main_loop
    await ist.connect()
  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/a_inverter.py", line 103, in connect
    if not await self.read_sensors(sensors=SOPT.startup):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ha_addon_sunsynk_multi/a_inverter.py", line 59, in read_sensors
    await self.inv.read_sensors(sensors)
  File "/usr/local/lib/python3.11/site-packages/sunsynk/sunsynk.py", line 134, in read_sensors
    raise IOError("; ".join(errs))
OSError: OSError reading 5 registers from 3: Failed to read register 3; OSError reading 2 registers from 20: Failed to read register 20; timeout reading 1 registers from 117; timeout reading 6 registers from 148
17:53:20 ERROR   Task was destroyed but it is pending!
task: <Task pending name='ConnKeeper' coro=<PySolarmanV5Async._conn_keeper() running at /usr/local/lib/python3.11/site-packages/pysolarmanv5/pysolarmanv5_async.py:139> wait_for=<Future pending cb=[Task.task_wakeup()]>>
[17:53:20] WARNING: Halt add-on with exit code 1
s6-rc: info: service legacy-services: stopping
```

**Reply by Pablo-or-Pablo** [reply]
Full YAML:

```yaml
DRIVER: solarman
INVERTERS:
  - SERIAL_NR: "2302278016"
    HA_PREFIX: SS
    MODBUS_ID: 1
    DONGLE_SERIAL_NUMBER: "1078800619"
    PORT: tcp://192.168.1.182:8899
SENSOR_DEFINITIONS: three-phase
SENSORS:
  - energy_management
  - power_flow_card
SENSORS_FIRST_INVERTER:
  - settings
MANUFACTURER: Deye
READ_ALLOW_GAP: 5
READ_SENSORS_BATCH_SIZE: 60
SCHEDULES:
  - KEY: W
    READ_EVERY: 10
    REPORT_EVERY: 60
    CHANGE_ANY: false
    CHANGE_BY: 80
    CHANGE_PERCENT: 0
NUMBER_ENTITY_MODE: auto
MQTT_HOST: core-mosquitto
MQTT_PORT: 1883
MQTT_USERNAME: xxxxx
MQTT_PASSWORD: xxxxx
DEBUG: 0
DEBUG_DEVICE: >-
  /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_20afef646093eb11b8a11b4f3d98b6d1-if00-port0
```

**Reply by Xhanti** [reply]
@Pablo-or-Pablo my bad I pointed at the wrong value. when you visit http://192.168.1.182/index_cn.html (adapted for you, login is admin admin), do you see 1078800619 at position B, not position A

<img width="440" alt="image" src="https://github.com/kellerza/sunsynk/assets/2096304/158a73ae-3a79-4adf-acef-9a5b9c0baf01">

**Comment by kellerza** [comment]
You should also set `READ_SENSORS_BATCH_SIZE: ` lower. (although the error does not seem directly related to this)

**Reply by Pablo-or-Pablo** [reply]
OK. I set this parametr to 20.
And I understand my fault - please correct manual, because dongle is the name of USB for MQTT.
In DEYE app it is called "logger". So serial of Inverter and serial of Logger - it shlould be easier to understand.

Also I have still some errors:
20:32:33 INFO    MQTT: Connection successful
20:32:34 INFO    Read every (inverter 1)
+-----+--------------------------------------------------------------------------------+
|  s  |                                    Sensors                                     |
+-----+--------------------------------------------------------------------------------+
|  5  | prog2_time, prog2_charge, prog1_capacity, prog5_charge, battery_low_capacity,  |
|     |      prog4_charge, prog4_capacity, load_limit, prog3_power, prog1_power,       |
|     |  priority_load, prog2_capacity, prog5_capacity, prog3_capacity, prog3_charge,  |
|     | prog5_power, prog6_capacity, prog1_time, prog4_time, prog2_power, prog3_time,  |
|     |  prog5_time, use_timer, prog6_charge, prog4_power, prog1_charge, prog6_time,   |
|     |                                  prog6_power                                   |
|  10 |      grid_power, pv1_power, inverter_power, battery_power, grid_ct_power       |
|  15 |     overall_state, pv1_current, pv1_voltage, battery_soc, grid_frequency,      |
|     |                 grid_voltage, battery_current, battery_voltage                 |
| 300 | day_battery_discharge, day_grid_import, total_battery_charge, total_pv_energy, |
|     |    day_load_energy, total_grid_import, day_grid_export, day_battery_charge,    |
|     |           total_battery_discharge, total_grid_export, day_pv_energy            |
+-----+--------------------------------------------------------------------------------+
20:32:34 INFO    Report every (inverter 1)
+-----+--------------------------------------------------------------------------------+
|  s  |                                    Sensors                                     |
+-----+--------------------------------------------------------------------------------+
|  60 |      grid_power, pv1_power, inverter_power, battery_power, grid_ct_power       |
| 300 | prog2_time, overall_state, prog2_charge, prog1_capacity, total_battery_charge, |
|     |   prog5_charge, battery_low_capacity, total_battery_discharge, prog4_charge,   |
|     |   prog4_capacity, total_grid_export, load_limit, prog3_power, grid_voltage,    |
|     |   battery_current, pv1_current, prog1_power, priority_load, prog2_capacity,    |
|     | total_pv_energy, prog5_capacity, day_load_energy, prog3_capacity, battery_soc, |
|     |   prog3_charge, prog5_power, day_battery_charge, prog6_capacity, prog1_time,   |
|     |   prog4_time, battery_voltage, day_grid_import, pv1_voltage, grid_frequency,   |
|     |     prog2_power, prog3_time, prog5_time, use_timer, day_battery_discharge,     |
|     |  prog6_charge, prog4_power, total_grid_import, day_grid_export, prog1_charge,  |
|     |                     prog6_time, day_pv_energy, prog6_power                     |
+-----+--------------------------------------------------------------------------------+
20:33:06 INFO    Frame error retry attempt(0): V5 frame contains invalid sequence number
20:33:10 INFO    Frame error retry attempt(0): V5 frame contains invalid sequence number
20:33:13 ERROR   OSError in read_ss: timeout reading 1 registers from 500 (poll_need_to_read)
....

And last thing - why time is -2h? It should be 22 not 20.
